GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -3117,5 +3119,56 @@ public void PopulateFavouriteRepositoriesMenu(ToolStripDropDownItem container, i
                 _form.PopulateFavouriteRepositoriesMenu(container, repositoryHistory);
             }
         }
+
+        private void WireDragDrop(Control.ControlCollection ctls)
+        {
+            foreach (Control ctl in ctls)
+            {
+                ctl.AllowDrop = true;
+                ctl.DragEnter += FormBrowse_DragEnter;
+                ctl.DragDrop += FormBrowse_DragDrop;
+                WireDragDrop(ctl.Controls);
```

Comment:
Instead of recursion it is probably better to use a breadth first traversal (using a `Queue<Control>`).
Besides we have `FindDescendants` method that retrieves all children controls:
https://github.com/gitextensions/gitextensions/blob/55bf8dfb294486a247f0a60859c5f1cba1d904cf/GitExtUtils/GitUI/ControlUtil.cs#L14-L34


I would also move the functionality into `OnLoad` method instead creating a separate method, primarily for the reasons of safety - we don't want someone call this method by accident.